### ISSUE TITLE
log paths not validated

### ISSUE THREAD

**Issue by kentcdodds** [kentcdodds/configuration-validator#1]
A really common use-case is when you are validating an object and there's a misspelling. Take this for example:

``` javascript
const validators = [
  {key: 'foo.bar.baz', validate: () => 'foo error', description: 'foo bar baz thing'},
]
const config = {foo: {bar: {baaz: 'error stuff'}}}
const result = checkConfig(config, validators)
```

There is an error in the spelling of `baaz`, however currently this will not log anything because there is no validator covering that misspelling.

What I think I want to have happen is for us to develop a function that can generate all possible paths for properties in a given object, and then another function that can take an array of paths and find the paths that are not covered. Then we could log a warning that there are some paths that are not covered with validation.

The developer would be able to turn off this warning for specific paths or all paths if they choose, but I think that the default should be to warn to save people trouble from misspellings. Instructions should be kept in a new file in the repo in `other/` called `WARNINGS_AND_ERRORS.md` and we could link the user to that file (using the specific tagged version in the URL to maintain history). I did this with angular-formly and it was quite helpful.

Definitely would love to see someone work on this!

**Comment by kentcdodds** [comment]
I've added tests for one of the functions already. See the npm scripts to run these.
